### ISSUE TITLE
Install Foxx services in arangodb3 (beta) using docker (Mac)

### ISSUE THREAD

**Issue by ifcologne** [arangodb/arangodb#1844]
I'm trying to install a Foxx service from Github but it doesn't work as expected.

I don't get a response from the UI and neither the service is created. But if I try again, clicking on install, at least I get an error 500 as there is already a service on this mount point.

```
Updated local repository information with 28 service(s)
Updated local repository information with 28 service(s)
Updated local repository information with 28 service(s)
2016-05-30T23:04:57Z [1] ERROR Error: An service is already installed at this location.
2016-05-30T23:04:57Z [1] ERROR     at _install (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/manager.js:1048:11)
2016-05-30T23:04:57Z [1] ERROR     at Object.install (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/manager.js:1108:17)
2016-05-30T23:04:57Z [1] ERROR     at Middleware.<anonymous> (/usr/share/arangodb3/js/apps/system/_admin/aardvark/APP/foxxes.js:92:29)
2016-05-30T23:04:57Z [1] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:366:15)
2016-05-30T23:04:57Z [1] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:362:7)
2016-05-30T23:04:57Z [1] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:362:7)
2016-05-30T23:04:57Z [1] ERROR     at /usr/share/arangodb3/js/node/node_modules/lodash/index.js:7363:25
2016-05-30T23:04:57Z [1] ERROR     at Middleware.router.use (/usr/share/arangodb3/js/apps/system/_admin/aardvark/APP/foxxes.js:49:3)
2016-05-30T23:04:57Z [1] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:366:15)
2016-05-30T23:04:57Z [1] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:362:7)
```

![screen shot 2016-05-31 at 01 05 53](https://cloud.githubusercontent.com/assets/2922081/15659505/eb3f253a-26cb-11e6-9799-e1b37a16096c.png)

After that the UI get's unresponsive (in many cases) and shutting down the server leads to crashing V8 contexts.

```
2016-05-30T23:25:09Z [1] INFO control-c received, beginning shut down sequence
2016-05-30T23:25:15Z [1] WARNING sending termination signal to V8 context
2016-05-30T23:25:15Z [1] WARNING sending termination signal to V8 context
2016-05-30T23:25:15Z [1] WARNING sending termination signal to V8 context
2016-05-30T23:25:15Z [1] WARNING sending termination signal to V8 context
2016-05-30T23:25:15Z [1] WARNING sending termination signal to V8 context
2016-05-30T23:26:15Z [1] FATAL cannot shutdown V8 contexts
```

I'm using ArangoDB 3.0.x-devel [linux] 64bit, using VPack 0.1.30, ICU 54.1, V8 5.0.71.39 on Docker for Mac (1.11.1-beta13.1 (build: 8193)).

EDIT (pluma): formatted log to be more readable

**Comment by pluma** [comment]
Could you clarify what service you were trying to install? By "from GitHub" do you mean from the store or from a GitHub repository directly?

**Comment by pluma** [comment]
I've adjusted the store logic to properly patch services with the metadata from the store (e.g. to override the compatibility info). All services other than hello-foxx (which uses CoffeeScript) should now be installable, although some may not yet work properly in legacy mode. I've also changed the error handling for when a service already exists so a proper status code is returned by aardvark (CONFLICT instead of 500).

I'm not sure what is causing the behaviour you are describing, however.
